pkg/template/v2/pem.go

```diff
@@ -56,6 +56,57 @@ func filterPEM(pemType, input string) (string, error) {
 	return string(blocks), nil
 }
 
+// getCertificates - returns a list of certs from a pem which may contain multiple certs.
+func getCertificates(input string) ([]string, error) {
+	certs := make([]string, 0)
+
+	data := []byte(input)
+	var blocks []byte
+	var block *pem.Block
+	var rest []byte
+	for {
+		block, rest = pem.Decode(data)
+		data = rest
+
+		if block == nil {
+			break
+		}
+
+		var buf bytes.Buffer
+		err := pem.Encode(&buf, block)
+		if err != nil {
+			return certs, err
+		}
+		blocks = append(blocks, buf.Bytes()...)
+		certs = append(certs, buf.String())
+	}
+
+	if len(blocks) == 0 && len(rest) != 0 {
+		return certs, errors.New(errJunk)
+	}
+
+	return certs, nil
+}
+
+// filterPEMChain - returns the server certificate.
+func filterPEMChain(input string) (string, error) {
+	certs, err := getCertificates(input)
+	if err != nil {
+		return "", err
+	}
+	return certs[0], nil
+}
+
+// filterPEMServer - returns the intermediate certificates.
+func filterPEMServer(input string) (string, error) {
+	certs, err := getCertificates(input)
+	if err != nil {
+		return "", err
+	}
+	// return all but the first
+	return strings.Join(certs[1:], "\n"), nil
+}
+
```

Comment:
You're making the assumption that certificates are already ordered. From my experience, this is not something we can or should take for granted.

For context, sometimes you have a more complex chain, e.g. `leaf -> intermediate -> intermediate -> root`, multiple chains or adjacent certificates. We should also take that into account.

Note on Naming:  `filterPEMChain` and `filterPEMServer` - looking at the function name it is not immediately clear what the function does. `filterPEMChain` returns the first certificate, `filterPEMServer` returns all but the first. If the order of certificates is not ordered from `leaf` to `root`, then we return unexpected results. 
The leaf certificate could also be a client or peer certificate or something hand cranked with custom X.509 extensions.

I suggest to:
1. create a function which attempts to order the certificates. Could be that there are multiple chains or none at all.
2. change the naming that resembles the behaviour: i suggest to build it around `leaf` and `chain`. It this also allows us to add / compose functions later on to filter for certificate types such as server/client/peer/intermediate-ca/root-ca certificates